chopper/lib/src/base.dart

```diff
@@ -144,12 +147,14 @@ class ChopperClient {
         res = await i.onResponse(res);
       } else if (i is ResponseInterceptorFunc) {
         res = await i(res);
```

Comment:
passing the body type here should be enough
`i<BodyType>(res)`

Reply:
Not sure what you mean here, passing the body type is not needed

Reply:
if you don't pass it will always be dynamic, the interceptor will never not the body type of the response

Reply:
The interceptor knows the type since it is inferred by dart from the response that is given (since they have the same body type value)